From 32ef76d592e18292e3ab019817084eabf131aa44 Mon Sep 17 00:00:00 2001
From: Zeng Heng <zengheng4@huawei.com>
Date: Thu, 26 Sep 2024 17:28:18 +0800
Subject: [PATCH 157/338] OPENEULER: arm64/mpam: Fix redefined reference of
 'mpam_detect_is_enabled'

hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/IATSCU

--------------------------------

Fix re-defined reference of 'mpam_detect_is_enabled' when CONFIG_ARM64_MPAM
is enabled.

Fixes: 797c68d3b52d ("arm64/mpam: Check mpam_detect_is_enabled() before accessing MPAM registers")
Signed-off-by: Zeng Heng <zengheng4@huawei.com>

Link: https://gitee.com/openeuler/kernel/commit/061f7a680d24eba2a09616cd90ab94a767cb95e3
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 arch/arm64/include/asm/cpufeature.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/arch/arm64/include/asm/cpufeature.h b/arch/arm64/include/asm/cpufeature.h
index a3907dc0effb..fc67e0f55cfc 100644
--- a/arch/arm64/include/asm/cpufeature.h
+++ b/arch/arm64/include/asm/cpufeature.h
@@ -873,14 +873,7 @@ static __always_inline bool system_supports_mpam_hcr(void)
 	return alternative_has_cap_unlikely(ARM64_MPAM_HCR);
 }
 
-#ifdef CONFIG_ARM64_MPAM
 bool mpam_detect_is_enabled(void);
-#else
-static inline bool mpam_detect_is_enabled(void)
-{
-	return false;
-}
-#endif
 
 int do_emulate_mrs(struct pt_regs *regs, u32 sys_reg, u32 rt);
 bool try_emulate_mrs(struct pt_regs *regs, u32 isn);
-- 
2.48.1

